### ISSUE TITLE
Upstreaming ffmpeg 4.2 or why not use system version

### ISSUE THREAD

**Issue by onny** [jellyfin/jellyfin-ffmpeg#6]
Hey,
i wonder why we have to relay on a custom, older ffmpeg version for Jellyfin and can not use the upstream or the one provided on the host system?
Some people have transcoding issues, e.g. here https://github.com/jellyfin/jellyfin/issues/1609 which might be already fixed upstream ...

Best regards,
Jonas

**Comment by joshuaboniface** [comment]
The TL;DR: We're working on it, but are limited by the old Emby code which breaks horribly when trying to use FFMPEG 4.1 or newer.

The longer version: Something in FFMPEG 4.1 broke the existing code such that a number of critical features, such as skipping around a video, break. The code that interfaces with FFMPEG is very fragile and very much spaghetti, so trying to clean it up has been something very few contributors have tried. We've made two attempts (one we abandoned, before making this self-distributed ffmpeg, and the other active at jellyfin/jellyfin#1434) to fix this, but in both cases we've ended up finding that we need to do a hard cut: either <4.1 is supported, or >=4.1 is supported, but not both.

The ideal solution is, of course, to rewrite the entire FFMPEG handler to work better, be more modular/consistent, and generally nicer. But this is a major undertaking that no one has volunteered to work on. So the next best thing is trying to support what we have as best we can.

This repo actually exists as a direct result of the first attempt at fixing this. We realized that having to rely on the distribution specific versions (all the way from 3.2 in Debian Stretch, the at-the-time Debian Stable, up to 4.1 in Arch, now 4.2) meant that we couldn't do the hard cut it seemed like we needed to do, so we packaged up 4.0 into our own package (`jellyfin-ffmpeg`) for Debuntu distributions and Docker, and just went from there, the intent being that we can then do a hard cut to a newer version for several different distribution releases at once if required.

As mentioned jellyfin/jellyfin#1434 attempts to support the newer versions, so we can upgrade this build to 4.2 once that's ready and tested, likely with our upcoming 10.4 release. But this is at least a month out. We don't have a better solution for transcoding issues until that time.

**Comment by onny** [comment]
Cool thanks for the info :+1:

**Comment by JustAMan** [comment]
As of Jellyfin 10.4.0 we support newer ffmpeg-s, and this repo has been updated to ffmpeg 4.2.1.
I think this closes the issue.
